src/odemis/driver/symphotime.py

```diff
@@ -0,0 +1,891 @@
+        if isinstance(decoded_msg, Message_EXPLAINED_STATUSMSG):
+            # Received a status message with an explanation string.
+            # Acknowledge it and log it.
+            self._sendMessage(Message_ENCODED_STATUSREPLY(decoded_msg.ecStatus))
+            logging.debug(decoded_msg.strExplanation)
+            err_status = decoded_msg.ecStatus
+
+            if err_status < 0:
+                self.state._set_value(HwError(decoded_msg.strExplanation), force_write=True)
+                self._measuring_flag.clear()
+                raise HwError("Error %d: %s" % (err_status, ERRCODE_DICT[err_status]))
+
+        elif isinstance(decoded_msg, Message_ENCODED_STATUSMSG):
+            # Received a status message with an encoded status
+            # Acknowledge it and log it
+
+            # Check status messages
+            err_status = decoded_msg.ecStatus
+            self._sendMessage(Message_ENCODED_STATUSREPLY(decoded_msg.ecStatus))
+
+            if err_status == PQ_ERRCODE_NO_ERROR:
+                # the measurement completed successfully.
+                self.state._set_value(model.ST_RUNNING, force_write=True)
+                    
+            elif err_status == PQ_ERRCODE_MEASUREMENT_READY:
+                # Measurement is complete
+                if self._measuring_flag.is_set():
+                    logging.info("Acquisition completed successfully.")
+                    self._measuring_flag.clear()
+                    # update metadata and send data array to subscribers
+                    md = self._metadata.copy()
+                    md[model.MD_DWELL_TIME] = self.scanner.dwellTime.value
+                    md[model.MD_ACQ_DATE] = self._tStart
+                    da = model.DataArray([[0]], md)
+                    self.data.notify(da)
+
+            else:
+                # All other status types denote errors. 
+                logging.error("Error %d: %s", err_status, ERRCODE_DICT[err_status])
+                self.state._set_value(HwError(ERRCODE_DICT[err_status]), force_write=True)
+                self._measuring_flag.clear()
+
+        elif isinstance(decoded_msg, Message_DATAFRAME_SRVNACK):
+            # A measurement data frame was received from the server
+            # Do not respond to this type.
+            if not self._measuring_flag.is_set():
+                # We should not receive this message if not measuring
+                raise HwError("Measurement data frame received while not measuring. ")
+
+            # convert the OptionalDataRecord type to a regular python dictionary for logging
+            data = {r.name : r.data for r in decoded_msg.odOptional.values()}
+            logging.info("Acq: %s", data)
+
+            # update metadata and vigilant attributes based on data coming back
+            # from the server
+            if "ServerVersion" in data:
+                self._metadata[model.MD_SW_VERSION] = data['ServerVersion']
+            if "ResultingFilename" in data:
```

Comment:
Symphotime will send back the filename it actually uses (it adds an extension and all). I added this so that I could make a testcase to make sure the filename set correctly.

Reply:
OK, ok. I'm not super found of doing stuff only for test cases, but on the other hand, it's nice to be able to test the code! Leave it as is.